### ISSUE TITLE
Feature request: Multi-label classification support

### ISSUE THREAD

**Issue by hi-tech-jazz** [scikit-learn-contrib/hiclass#23]
It would be great to have support for multi-label prediction. Methods of achieving this is briefly mentioned in the Sillas and Freitas survey of local hierarchical patterns. Implementation inspiration could possibly be drawn from [this project](https://github.com/globality-corp/sklearn-hierarchical-classification), which shares its objective but unfortunately seems abandoned. Would this be possible?

**Comment by mirand863** [comment]
> 

Hello,

Thanks for the interest in HiClass!

Can you please tell me what is your use-case?

Do you need multi-label prediction for all 3 hierarchical classifiers or only on the local classifier per node would be enough? I am currently working on updating the local classifier per level to pass sklearn's checks, but the next feature I will add is probably multi-label classification at least for the local classifier per node.

**Comment by hi-tech-jazz** [comment]
Nice to hear! It would be valuable for all, as I am trying to compare each method. The application is to annotate topical categories to blog posts.

**Comment by mirand863** [comment]
> Nice to hear! It would be valuable for all, as I am trying to compare each method. The application is to annotate topical categories to blog posts.

@hi-tech-jazz do you happen to know of any public dataset similar to yours that I could use for testing?

**Comment by hi-tech-jazz** [comment]
> > Nice to hear! It would be valuable for all, as I am trying to compare each method. The application is to annotate topical categories to blog posts.
> 
> @hi-tech-jazz do you happen to know of any public dataset similar to yours that I could use for testing?

Sorry for replying late. [This dataset](https://www.kaggle.com/datasets/kashnitsky/hierarchical-text-classification) is pretty similar to mine.

**Comment by mirand863** [comment]
Hi @hi-tech-jazz,

I am not sure this dataset is appropriate for multi-label classification. Furthermore, I have talked with my supervisor and we decided multi-label classification will not be implemented anytime soon. Firstly, I have no proper dataset or metrics to test any implementation. Secondly, there are other features and documentation updates that we judged more important for a re-submission to JMLR. However, I will keep this issue open and revisit it in the future if circumstances change.

Kind regards,
Fabio

**Comment by mirand863** [comment]
Hello @hi-tech-jazz,

Just a quick update. I found a dataset to test multi-label classification. I will be working on this shortly and keep you posted.

Kind Regards,
Fabio

**Comment by mirand863** [comment]
> > > Nice to hear! It would be valuable for all, as I am trying to compare each method. The application is to annotate topical categories to blog posts.
> > 
> > 
> > @hi-tech-jazz do you happen to know of any public dataset similar to yours that I could use for testing?
> 
> Sorry for replying late. [This dataset](https://www.kaggle.com/datasets/kashnitsky/hierarchical-text-classification) is pretty similar to mine.

Hello @hi-tech-jazz,
Sorry for taking too long to start working on this. We are deciding on ways to represent the multi-label hierarchical data for training and predicting.
So far we are thinking about adding another dimension, for example:
```
[
    [
        [Mermaid], [Fish, Woman]
    ],
    [
        [Platypus], [Duck, Beaver, Otter]
    ],
]
```
**Do you have any suggestions that would make your life easier as a user?**

**Comment by RobinRojowiec** [comment]
I have a similar problem. Wanted to have multiple labels on the leaves of the hierarchy tree. Your solution looks exactly like what I was looking for.
Hope that is added soon.

**Comment by mirand863** [comment]
Hi @RobinRojowiec 

Thank you for the interest! There is an open pull request where we started a multi-label implementation for the local classifiers per node and per parent node https://github.com/scikit-learn-contrib/hiclass/pull/76. While it is still far from being marked as complete for merging, perhaps you could give it a try and let me know if you have any issues or feedback.

Best regards,
Fabio
